ldelf/ta_elf_rel.c

```diff
@@ -35,19 +35,26 @@ static uint32_t elf_hash(const char *name)
 static bool __resolve_sym(struct ta_elf *elf, unsigned int st_bind,
 			  unsigned int st_type, size_t st_shndx,
 			  size_t st_name, size_t st_value, const char *name,
-			  vaddr_t *val)
+			  vaddr_t *val, bool weak_ok, bool *found_weak_undef)
```

Comment:
It seems that `found_weak_undef` isn't really needed. Each time this is set to `true` `*val` is assigned `0` by the caller.

Reply:
True, I will simplify.